src/main/java/com/socialmedia/service/PostService.java

```diff
@@ -64,6 +65,13 @@ public Post delete(Long id) {
 
   @Override
   public Post update(Post existingEntity, Post incomingEntity) {
+    incomingEntity.setDate(System.currentTimeMillis());
+    if (incomingEntity.getImage() != null) {
+      Image postImage = imageService.getById(incomingEntity.getImage().getId());
```

Comment:
You may also check if incoming image id != existing not to go to DB for same img